pkg/domain/infra/tunnel/farm.go

```diff
@@ -0,0 +1,102 @@
+package tunnel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"sync"
+
+	istorage "github.com/containers/image/v5/storage"
+	"github.com/containers/podman/v4/pkg/bindings/system"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+)
+
+const (
+	remoteFarmImageBuilderDriver = "podman-remote"
+)
+
+// FarmName returns the remote engine's name.
+func (ir *ImageEngine) FarmName(ctx context.Context) string {
+	return ir.Farm
+}
+
+// FarmDriver returns a description of the image builder driver
+func (ir *ImageEngine) FarmDriver(ctx context.Context) string {
+	return remoteFarmImageBuilderDriver
+}
+
+func (ir *ImageEngine) fetchInfo(_ context.Context) (os, arch string, nativePlatforms []string, err error) {
+	engineInfo, err := system.Info(ir.ClientCtx, &system.InfoOptions{})
+	if err != nil {
+		return "", "", nil, fmt.Errorf("retrieving host info from %q: %w", ir.Farm, err)
+	}
+	os = engineInfo.Host.OS
+	arch = engineInfo.Host.Arch
+	nativePlatform := os + "/" + arch // TODO: pester someone about returning variant info
+	return os, arch, []string{nativePlatform}, nil
+}
+
+// FarmInspect returns information about the remote engines in the farm
+func (ir *ImageEngine) FarmInspect(ctx context.Context) (*entities.FarmInspectReport, error) {
+	var (
+		platforms          sync.Once
+		platformsErr       error
+		os, arch, variant  string
+		nativeP, emulatedP []string
+	)
+	platforms.Do(func() {
```

Comment:
It looks like `platforms.Do()` is always going to call the passed-in function.

Reply:
fixed